File: 2022/src/day02.rs
fn part1(value: &str) -> i32 {
    match value {
        "A X" => 4,
        "A Y" => 8,
        "A Z" => 3,
        "B X" => 1,
        "B Y" => 5,
        "B Z" => 9,
        "C X" => 7,
        "C Y" => 2,
        "C Z" => 6,
        &_ => 0,
    }
}

fn part2(value: &str) -> i32 {
    match value {
        "A X" => 3,
        "A Y" => 4,
        "A Z" => 8,
        "B X" => 1,
        "B Y" => 5,
        "B Z" => 9,
        "C X" => 2,
        "C Y" => 6,
        "C Z" => 7,
        &_ => 0,
    }
}

pub fn run(input: &str) -> String {
    let mut score_part1: i32 = 0;
    let mut score_part2: i32 = 0;

    for line in input.lines() {
        score_part1 += part1(line);
        score_part2 += part2(line);
    }

    format!("part1: {:?}\npart2: {:?}", score_part1, score_part2)
}

File: 2023/src/day02.swift
import Foundation

struct Day02 {
    func run(input: [String]) {
        var part1 = 0
        var part2 = 0
        
        input.forEach { line in
            let game = line.split(separator: ":")
            var colors = (red: 0, green: 0, blue: 0)
            
            game[1].split(separator: ";").forEach { set in
                set.split(separator: ",").forEach { cube in
                    let c = cube.split(separator: " ")
                    
                    switch c[1] {
                    case "red":     colors.red      = max(Int(c[0])!, colors.red)
                    case "green":   colors.green    = max(Int(c[0])!, colors.green)
                    case "blue":    colors.blue     = max(Int(c[0])!, colors.blue)
                    default:        ()
                    }
                }
            }
            
            if !(colors.red > 12 || colors.green > 13 || colors.blue > 14) {
                part1 += Int(game[0].split(separator: " ")[1])!
            }
            
            part2 += (colors.red * colors.green * colors.blue)
        }
        
        print("part1: \(part1)")
        print("part2: \(part2)")
    }
}
